xbmc/application/Application.cpp

```diff
@@ -2331,11 +2331,27 @@ bool CApplication::PlayFile(CFileItem item,
   if (PLAYLIST::IsPlayList(item))
     return false;
 
-  // Translate/Resolve the url if needed
-  const std::unique_ptr<IDirectory> dir{CDirectoryFactory::Create(item)};
-  if (dir && !dir->Resolve(item))
+  // Translate/Resolve the url if needed - recursively, but only limited times.
+  static constexpr size_t MAX_ITEM_RESOLVE_ATTEMPTS{5};
```

Comment:
Maybe we can place this somewhere else? I think the recursive item resolution is set to 5 in other places too

Reply:
Yeah, also used in filesystem/PluginDirectory.cpp - I think we move this constant to filesystem/IDirectory.h and use it here and for plugins.